### PR TITLE
[AGENT-5439] add model downloader sources

### PR DESCRIPTION
**Pull request by alexromanyuk**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

### CODE REVIEW

File: public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+        self._deployment_id = deployment_id
+        self._dr_client = dr.Client()
+        self._mlops_service_url = os.getenv("DATAROBOT_ENDPOINT")
+        self._mlops_client = MLOpsClient(service_url=os.getenv("DATAROBOT_ENDPOINT").replace('/api/v2', ''),
+                                         api_key=os.getenv("DATAROBOT_API_TOKEN"),
+                                         verify=verify_ssl)
+        self._mlops = MLOps().set_deployment_id(self._deployment_id).set_api_spooler().init()
+
+        self._code_dir = code_dir
+        self._deployment_info = self._mlops_client.get_deployment(deployment_id)
```
Comment (elatt) [inline public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py:57]: Can this code work when there is no deployment? In Custom Model Testing there is no deployment but we still need to download model artifacts (and it's ok if we don't get progress updates in that case 🤷‍♂️ )

File: public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+
+
+class StoreURIPrefix:
+    S3 = "s3:"
+
+
+class ModelDownloader:
+    REMOTE_FILE_SUFFIX = ".remote"
+    METADATA_FILE = "model-metadata.yaml"
+    MLOPS_RUNTIME_PARAM_PREFIX = "MLOPS_RUNTIME_PARAM_"
```
Comment (elatt) [inline public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py:31]: We shouldn't do this. If we want to interact with runtime params, we need to use the standard Python abstraction.

File: public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+                                         verify=verify_ssl)
+        self._mlops = MLOps().set_deployment_id(self._deployment_id).set_api_spooler().init()
+
+        self._code_dir = code_dir
+        self._deployment_info = self._mlops_client.get_deployment(deployment_id)
+        self._model_package_id = self._deployment_info[MLOpsClient.RESPONSE_MODEL_PACKAGE_KEY][
+            MLOpsClient.RESPONSE_MODEL_PACKAGE_ID_KEY
+        ]
+        pprint.pprint(self._deployment_info)
+        print("model package id: {}".format(self._model_package_id))
```
Comment (elatt) [inline public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py:62]: We should use logging module.

File: public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py (hunk 2 above)
Comment (alexromanyuk) [reply public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py:31]: Maybe we'll skip this library for this release altogether? It's not ready yet.

File: public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py [hunk 4]
```diff
@@ -0,0 +1,542 @@
...
+        model_zip = None
+        for zip_filename in zip_content:
+            if zip_filename.startswith("model") and zip_filename.endswith(".zip"):
+                model_zip = zip_filename
+        print("model zip filename: {}".format(model_zip))
+        if model_zip is None:
+            raise Exception("Error could not find model zip file: {}".format(zip_content))
+
+        print("extracting")
+        os.chdir(output_dir)
```
Comment (elatt) [inline public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py:136]: Yuk, library code that `chdir`'s. We should use relative paths instead.

File: public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py (hunk 2 above)
Comment (alexromanyuk) [reply public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py:31]: @elatt I'm going to close this PR, I feel that we have to skip this library from the current release and add it when it's covered with tests, have it's own pipeline to be published, etc.

Users have ability to run the `custom.py` load_model hook.

File: public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py [hunk 5]
```diff
@@ -0,0 +1,542 @@
...
+    @staticmethod
+    def _handle_credentials_param(param_name):
+        param_json = os.environ.get(param_name, None)
+        if param_json is None:
+            raise EnvironmentError("expected environment variable '{}' to be set".format(param_name))
+        print(f"param_json: {param_json}")
+
+        json_content = json.loads(param_json)
+        if param_json is None:
+            raise EnvironmentError("expected environment variable '{}' to be json".format(param_name))
```
Comment (elatt) [inline public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py:83]: We should be using the DRUM class to do all this

File: public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py (hunk 2 above)
Comment (elatt) [reply public_dropin_gpu_environments/nemollm_inference_ms/artifact_downloader/datarobot_custom_code/model_artifact_downloader.py:31]: > Maybe we'll skip this library for this release altogether? It's not ready yet.

Sounds fine with me 🤷‍♂️

**Pull request closed without merging**